src/main/java/org/medicmobile/webapp/mobile/FilePickerHandler.java

```diff
@@ -0,0 +1,163 @@
+package org.medicmobile.webapp.mobile;
+
+import static android.app.Activity.RESULT_OK;
+import static android.provider.MediaStore.ACTION_IMAGE_CAPTURE;
+import static android.provider.MediaStore.ACTION_VIDEO_CAPTURE;
+import static android.provider.MediaStore.Audio.Media.RECORD_SOUND_ACTION;
+import static android.provider.MediaStore.EXTRA_OUTPUT;
+import static org.medicmobile.webapp.mobile.EmbeddedBrowserActivity.RequestCode;
+import static org.medicmobile.webapp.mobile.MedicLog.trace;
+import static org.medicmobile.webapp.mobile.MedicLog.warn;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Environment;
+import android.webkit.ValueCallback;
+import android.webkit.WebChromeClient.FileChooserParams;
+
+import androidx.core.content.FileProvider;
+
+import java.io.File;
+import java.util.Arrays;
+
+public class FilePickerHandler {
+
+	private final Activity context;
+	private ValueCallback<Uri[]> filePickerCallback;
+	private Uri tempFileUri;
+
+	public FilePickerHandler(Activity context) {
+		this.context = context;
+	}
+
+	void setFilePickerCallback(ValueCallback<Uri[]> filePickerCallback) {
+		this.filePickerCallback = filePickerCallback;
+	}
+
+	void setTempFileUri(Uri tempFileUri) {
+		this.tempFileUri = tempFileUri;
+	}
+
+	void openPicker(FileChooserParams fileChooserParams, ValueCallback<Uri[]> filePickerCallback) {
+		trace(this, "FilePickerHandler :: Start file capture activity");
+		setFilePickerCallback(filePickerCallback);
+		setTempFileUri(null);
+		startFileCaptureActivity(fileChooserParams);
+	}
+
+	void processResult(int resultCode, Intent intent) {
+		if (resultCode != RESULT_OK) {
+			String message = "FilePickerHandler :: Bad result code: %s. " +
+				"Either the user didn't select a file or there's an error during the operation.";
+			warn(this, message, resultCode);
+			sendDataToWebapp(null);
+			return;
+		}
+
+		sendDataToWebapp(intent);
+	}
+
+//> PRIVATE
+
+	/**
+	 * Executes the file picker callback.
+	 * The callback will give control back to WebView that automatically fetches the file by using a
+	 * provided URI, finally it assigns the bytes to the form's input. Even if there's not URI (ie null),
+	 * it needs to give control back to WebView, otherwise when clicking again on the form's input
+	 * it won't open the file picker.
+	 *
+	 * @param intent Intent from activity's response.
+	 */
+	private void sendDataToWebapp(Intent intent) {
+		if (this.filePickerCallback == null) {
+			warn(this, "FilePickerHandler :: Cannot send data back to webapp, filePickerCallback is null.");
+			return;
+		}
+
+		Uri uri = getSelectedFileUri(intent);
+		trace(this, "FilePickerHandler :: Sending data back to webapp, URI: %s", uri);
+		this.filePickerCallback.onReceiveValue(uri == null ? null : new Uri[] { uri });
```

Comment:
To add more info: I just double checked about sending `null` in `onReceiveValue`, this is the [recommended way](https://developer.android.com/reference/android/webkit/WebChromeClient#onShowFileChooser(android.webkit.WebView,%20android.webkit.ValueCallback%3Candroid.net.Uri[]%3E,%20android.webkit.WebChromeClient.FileChooserParams)) in the android site.